packages/ods-react/src/components/input/src/contexts/useInput.tsx

```diff
@@ -0,0 +1,125 @@
+import type { Locale } from '../../../../utils/locales';
+import type { InputMaskState } from '../constants/input-mask-state';
+import { type JSX, type ReactNode, createContext, useContext } from 'react';
+
+interface InputRootProp {
+  /**
+   * Whether the clear button is displayed.
+   */
+  clearable?: boolean,
+  /**
+   * Custom aria-label for the clear button.
+   */
+  clearButtonAriaLabel?: string,
+  /**
+   * Whether the component is in error state.
+   */
+  invalid?: boolean,
+  /**
+   * Custom i18n labels.
+   */
+  i18n?: Record<string, string>,
+  /**
+   * Whether the component is in loading state.
+   */
+  loading?: boolean,
+  /**
+   * Whether the masked display is active and its initial state.
+   */
+  maskOption?: {
+    enable: boolean,
+    initialState?: InputMaskState,
+  },
+  /**
+   * Custom aria-label for the mask toggle button.
+   */
+  maskButtonAriaLabel?: string,
+  /**
+   * Locale for i18n labels.
+   */
+  locale?: Locale,
+  /**
+   * Callback fired when the input value is cleared.
+   */
+  onClear?: () => void,
+  /**
+   * Custom aria-label for the search button.
+   */
+  searchButtonAriaLabel?: string,
+  // 'type' retiré pour éviter le conflit avec les props HTML
+}
+
+type InputContextType = Pick<InputRootProp, 'i18n' | 'locale'>;
+
+interface InputProviderProp extends InputContextType {
+  children: ReactNode,
+}
+
+const InputContext = createContext<InputContextType>({});
+
+function InputProvider({ children, i18n, locale }: InputProviderProp): JSX.Element {
+  return (
+    <InputContext.Provider value={{
+      i18n,
+      locale,
+    }}>
+      { children }
+    </InputContext.Provider>
+  );
+}
+
+function useInput(): InputContextType {
+  return useContext(InputContext);
+}
+
+interface InputActionsProps {
```

Comment:
you can define it on the componet directly if not shared anywhere, plus it is an internal component, you don't need explicit documentation